cartography/intel/aws/redshift.py

```diff
@@ -10,11 +10,111 @@
 from cartography.util import run_cleanup_job
 from cartography.util import timeit
 from cloudconsolelink.clouds.aws import AWSLinker
+from botocore.exceptions import ClientError
 
 logger = logging.getLogger(__name__)
 aws_console_link = AWSLinker()
 
 
+@timeit
+@aws_handle_regions
+def get_redshift_reserved_node(boto3_session: boto3.session.Session, region: str, current_aws_account_id: str) -> List[Dict]:
+    try:
+        client = boto3_session.client('redshift', region_name=region)
+        paginator = client.get_paginator('describe_reserved_nodes')
+        reserved_nodes: List = []
+        for page in paginator.paginate():
+            reserved_nodes.extend(page['ReservedNodes'])
+        for reserved_node in reserved_nodes:
+            reserved_node['region'] = region
+            reserved_node['arn'] = f"arn:aws:redshift:{region}:{current_aws_account_id}:reserved-node/{reserved_node['ReservedNodeId']}"
+        return reserved_nodes
+
+    except ClientError as e:
+        logger.error(f'Failed to call redshift describe_reserved_nodes: {region} - {e}')
+        return reserved_nodes
+
+
+def load_redshift_reserved_node(session: neo4j.Session, reserved_nodes: List[Dict], current_aws_account_id: str, aws_update_tag: int) -> None:
+    session.write_transaction(_load_redshift_reserved_node_tx, reserved_nodes, current_aws_account_id, aws_update_tag)
+
+
+@timeit
+def _load_redshift_reserved_node_tx(
+    tx: neo4j.Transaction, data: List[Dict], current_aws_account_id: str,
+    aws_update_tag: int,
+) -> None:
+    ingest_rds_secgroup = """
+    UNWIND {reserved_nodes} as reserved_node
+        MERGE (node:RedshiftReservedNode{id: reserved_node.arn})
```

Comment:
Is it possible to build a relationship between Redshift Cluster or Instance with Reserved Node?

Reply:
Reserved Node Response doesn’t contain any information about Redshift Cluster or Instance
`"ReservedNodes": [
      {
         "OfferingType": "Heavy Utilization",
         "FixedPrice": "",
         "NodeType": "dw.hs1.xlarge",
         "ReservedNodeId": "1ba8e2e3-bc01-4d65-b35d-a4a3e931547e",
         "UsagePrice": "",
         "RecurringCharges": [
            {
               "RecurringChargeAmount": "",
               "RecurringChargeFrequency": "Hourly"
            } ],
         "NodeCount": 1,
         "State": "payment-pending",
         "StartTime": "2013-02-13T17:08:39.051Z",
         "Duration": 31536000,
         "ReservedNodeOfferingId": "ceb6a579-cf4c-4343-be8b-d832c45ab51c"
      }
   ]`

Reply:
Redshift Clusters have a field called SourceReservedNodeId and TargetReservedNodeOfferingId. One of these could be used to build the relationship. But, i am not sure which one :)

Reply:
`SourceReservedNodeId` and `TargetReservedNodeOfferingId` both key haven’t proper description on api documentation page so i am little bit confused about this